compose/compiler/compiler-hosted/src/main/java/androidx/compose/compiler/plugins/kotlin/lower/decoys/CreateDecoysTransformer.kt

```diff
@@ -162,10 +229,112 @@ class CreateDecoysTransformer(
 
         originalFunctions += copied to declaration.parent
 
-        return original.apply {
+        original.apply {
             setDecoyAnnotation(newName.asString())
             stubBody()
         }
+
+        // We want to make origin constructor non-primary, cause ONLY primary constructor can have
+        // instance initializer call. We create a constructor's copy to transfer to non-primary,
+        // but after that, we should fix references to the origin one to make IR tree correct,
+        // modify by reflecting is the more convenient way :).
+        // If you want to copy from origin, should add @Decoy annotation on it also. References to
+        // the origin constructor haven't been changed, pass "SubstituteDecoyCallsTransformer"
+        // will check @Decoy annotation and change reference to decoy implementation.
+        if (context.platform.isNative() && original.isPrimary) {
```

Comment:
At what moment does it get broken if we keep `isPrimary` as is (not changing it here at all)?